internal/service/dns_config/api_auth_zone_resource.go

```diff
@@ -145,20 +153,28 @@ func (r *AuthZoneResource) Delete(ctx context.Context, req resource.DeleteReques
 
 	// Read Terraform prior state data into the model
 	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
-
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	httpRes, err := r.client.DNSConfigurationAPI.
-		AuthZoneAPI.
-		AuthZoneDelete(ctx, data.Id.ValueString()).
-		Execute()
-	if err != nil {
-		if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
-			return
+	err := retry.RetryContext(ctx, AuthZoneOperationTimeout, func() *retry.RetryError {
```

Comment:
Do you think we can make the timeout configurable? This will give us the option to set the lower timeouts as we don't need to wait the entirety of 2 minutes to delete an Auth zone in the prod cluster.

Reply:
The plugin will only wait for the whole 2 minutes if the sync hasn't happened, and most cases it will happen within seconds. I don't see why to make this configurable, as the user cannot control what happens in the production cluster. If you feel 2 minutes is too much, we can reduce it, but to me 2 minutes seem generous enough.